TUI: detect BSD vt console

justinmk opened the pull request:
FreeBSD console sets TERM=xterm, but it does not support xterm features
like cursor-shaping. GUI terminals typically set TERM=xterm-256color, so
on FreeBSD we can guess that that TERM=xterm is the degraded vt.

OpenBSD console sets TERM=vt220
https://github.com/openbsd/src/blob/master/etc/etc.amd64/ttys

NetBSD console sets TERM=vt100
https://github.com/NetBSD/src/blob/trunk/etc/etc.amd64/ttys

closes #8644

~~@vext01 is this also the case on OpenBSD?~~
~~@utkarsh009 is this also the case on NetBSD?~~

justinmk committed:
TUI: detect BSD vt console

FreeBSD console sets TERM=xterm, but it does not support xterm features
like cursor-shaping. GUI terminals typically set TERM=xterm-256color, so
on FreeBSD we can guess that TERM=xterm is the degraded vt.

OpenBSD console sets TERM=vt220
https://github.com/openbsd/src/blob/master/etc/etc.amd64/ttys

NetBSD console sets TERM=vt100
https://github.com/NetBSD/src/blob/trunk/etc/etc.amd64/ttys

closes #8644

utkarsh009 commented:
Nope. It’s vt220 in OpenBSD.
https://github.com/openbsd/src/blob/master/etc/etc.amd64/ttys
And vt100 in NetBSD:
https://github.com/NetBSD/src/blob/trunk/etc/etc.amd64/ttys


On Tue, 11 Dec 2018 at 6:50 AM, Justin M. Keyes <notifications@github.com>
wrote:

> FreeBSD console sets TERM=xterm, but it does not support xterm features
> like cursor-shaping. GUI terminals typically set TERM=xterm-256color, so
> on FreeBSD we can guess that that TERM=xterm is the degraded vt.
>
> This has so far only been reported for FreeBSD, but this patch assumes
> similar behavior for the other BSDs.
>
> closes #8644 <https://github.com/neovim/neovim/issues/8644>
>
> @vext01 <https://github.com/vext01> is this also the case on OpenBSD?
> @utkarsh009 <https://github.com/utkarsh009> is this also the case on
> OpenBSD?
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/neovim/neovim/pull/9347
> Commit Summary
>
>    - TUI: detect BSD vt console
>
> File Changes
>
>    - *M* src/nvim/tui/terminfo.c
>    <https://github.com/neovim/neovim/pull/9347/files#diff-0> (12)
>    - *M* src/nvim/tui/tui.c
>    <https://github.com/neovim/neovim/pull/9347/files#diff-1> (14)
>
> Patch Links:
>
>    - https://github.com/neovim/neovim/pull/9347.patch
>    - https://github.com/neovim/neovim/pull/9347.diff
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/neovim/neovim/pull/9347>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQUsvb0WihwngKW5CrgZp1ZUXWk8MvPZks5u3wh2gaJpZM4ZMfEm>
> .
>
-- 
Regards,
Utkarsh Anand

justinmk committed:
TUI: TERM=nsterm

Treat TERM=nsterm the same way in augment_terminfo() and
patch_terminfo_bugs().

ref #9244 463d28cc8079
